src/libraries/Common/src/System/Runtime/InteropServices/ComEventsMethod.cs

```diff
@@ -246,27 +260,38 @@ public void RemoveDelegates(Func<Delegate, bool> condition)
                     }
                 }
 
+                if (toRemove.Count == 0)
+                {
+                    // Nothing to remove
+                    return;
+                }
+
+                newWrappers = wrappers.GetRange(0, wrappers.Count);
                 foreach (int idx in toRemove)
                 {
-                    _delegateWrappers.RemoveAt(idx);
+                    newWrappers.RemoveAt(idx);
                 }
```

Comment:
If you simplified the storage such that you didn't bother with Delegate.Combine, then it seems like tthis loop could be simplified as well to something along the lines of:
```C#
List<DelegateWrapper> oldWrappers, newWrappers;
do
{
    oldWrappers = _delegateWrappers;
    newWrappers  = new List<DelegateWrapper>(oldWrappers);
    if (newList.RemoveAll(w => condition(w.Delegate)) == 0)
    {
        return;
    }
}
while (!PublishNewWrappers(newWrappers, oldWrappers));
```